hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/security/OzoneTokenIdentifier.java

```diff
@@ -77,6 +80,122 @@ public Text getKind() {
     return KIND_NAME;
   }
 
+  /** Instead of relying on proto serialization, this
+   *  provides  explicit serialization for OzoneTokenIdentifier.
+   * @return byte[]
+   */
+  public byte[] toUniqueSerializedKey() {
+    ByteBuffer result =
```

Comment:
Thanks @prashantpogde for working on this. The patch LGTM overall. Have a few questions:
Have you consider leverage the readFields from parent class to handle the basic non-protoc serialization which seems to be much simpler. 

```  
public byte[] toUniqueSerializedKey() throws IOException {
    DataOutputBuffer buf = new DataOutputBuffer();
    super.write(buf);
    WritableUtils.writeEnum(buf, getTokenType());
    // Set s3 specific fields.
    if (getTokenType().equals(S3AUTHINFO)) {
      WritableUtils.writeString(buf, getAwsAccessId());
      WritableUtils.writeString(buf, getSignature());
      WritableUtils.writeString(buf, getStrToSign());
    } else {
      WritableUtils.writeString(buf, getOmCertSerialId());
    }
    return buf.getData();
  }
```

Reply:
didn't try this. This looks simpler. Let me try this.

Reply:
I wanted to do explicit serialization. But this also looks safe serialization.